### PR TITLE
Set HCC_LAZYINIT=ON in caffe2 CI

### PR DESCRIPTION
**Pull request by bddppq**
Only meant for testing at this point, will consult amd folks before proceeding.

### CODE REVIEW

File: .jenkins/caffe2/test.sh [hunk 1]
```diff
@@ -8,6 +8,11 @@ if [[ "${BUILD_ENVIRONMENT}" == *-android* ]]; then
   exit 0
 fi
 
+if [[ $BUILD_ENVIRONMENT == *-rocm* ]]; then
+  # TODO: add this to docker image
+  export HCC_LAZYINIT=ON
```
Comment (houseroad) [inline .jenkins/caffe2/test.sh]: It's 'ON', not 1?

**Review by houseroad — APPROVED** [review]

File: .jenkins/caffe2/test.sh (hunk 1 above)
Comment (bddppq) [reply .jenkins/caffe2/test.sh]: Comes from amd folks https://github.com/pytorch/pytorch/pull/18640#issuecomment-489152269
Locally tried "ON", I believe it's on as long as the environment variable is set.

Comment (bddppq) [reply .jenkins/caffe2/test.sh]: lol https://github.com/RadeonOpenCompute/hcc/blob/e72465c/lib/mcwamp.cpp#L418-L423 it's literally testing for "ON" :-)

**Comment by bddppq** [comment]
@jithunnair-amd @petrex Is there any obviously downside of always setting HCC_LAZYINIT to true. We have observed significant startup time speedup, and also this could help unblock enabling pytorch multi-gpu tests on rocm.

**Comment by petrex** [comment]
@bddppq There is a plan to make `HCC_LAZYINIT` default in later ROCm; however I'd argue this would be the elixir to all the mgpu issues.

**Pull request closed without merging**